plugins/lookup/hashi_vault.py

```diff
@@ -321,6 +360,38 @@
 }
 
 
+# TODO: this is a workaround for deprecations not being shown in lookups
+# See: https://github.com/ansible/ansible/issues/73051
+# Fix: https://github.com/ansible/ansible/pull/73058
+#
+# If #73058 or another fix is backported, this should be removed.
+def deprecate(plugin_name='hashi_vault'):
+
+    # nicked from cli/__init__.py
+    # with slight customizations to help filter out relevant messages
+    # (must add `plugin_name:` to the `deprecation:` dict)
+
+    # warn about deprecated config options
+
+    for deprecated in list(C.config.DEPRECATED):
+        name = deprecated[0]
+        why = deprecated[1]['why']
+        if deprecated[1].get('plugin_name') != plugin_name:
```

Comment:
IMO I'd be better to distinguish by `collection_name` here, so you don't need to add `plugin_name`.

Reply:
yeah, I will do that in light of the doc validation change you mentioned,  just "lucky" that there's only one plugin in the collection at the moment, and hopefully some fix lands before we get to the point of a new plugin that also has deprecations 😅